sigmastate/src/main/scala/sigmastate/interpreter/ProverInterpreter.scala

```diff
@@ -556,7 +555,7 @@ trait ProverInterpreter extends Interpreter with ProverUtils {
     case ut: UnprovenTree => ut
 
     case t: ProofTree =>
-      log.warn("Wrong input in prove(): ", t);
+      logMessage(s"Wrong input in prove(): $t");
```

Comment:
Why unconditionally dumping to console instead of using flexible logger (with logging levels, different output options, configurable etc). Does not look like a good change.

Reply:
This is because there is no good dependency for logging which support Scala 2.11-2.13 and also Scala.js. 
Also sigma should be non-opinionated library so that it can be reused in different context (including the very constrained JS environment).
So the simplest way to abstract is just call some method and let the library users to override the default behavior.

Reply:
In this case better to extract logging functions into a separate interface likely

Reply:
This is out of this PR's scope. Created an issue https://github.com/ScorexFoundation/sigmastate-interpreter/issues/859

Reply:
the change is also out of PR scope seemingly, not corresponding to PR description. Consider to move it to corresponding PR

Reply:
Updated description